### ISSUE TITLE
Strange LAC/CID/PSC when on 2G EDGE

### ISSUE THREAD

**Issue by E3V3A** [SecUpwN/Android-IMSI-Catcher-Detector#254]
Setting my 4G (LTE) enabled phone to 2G (GSM-900) only, when using a network provider who normally operates on 3G (UMTS) the Android network icon shows "E" for *EDGE* calls only, and the LAC/CID and PSC are shown in the `Device Details` as:

> LAC: 1
> CID: 132
> PSC: 0

**Q:** Is this a standard for the "EDGE" network mode? 

If this is wrong, we should display a message to the user, informing him that his SIM is not compatible with 2G only mode...

Apparently: 

| icon | RAT | speed |
|:---- |:--- |:----- | 
G | GSM | up to 14 Kbps 
G | GPRS | 56 Kbps, steady
E | EDGE | up to 260 Kbps
3G | 3G (UMTS) | up to 380 Kbps
H | HSDPA | up to 7 Mbps
H+ | HSDPA+ | up to 28.8 Mbps

**Comment by He3556** [comment]
yes i have something similar. It looks like there are 1 or 2 digits missing in the LAC or CellID sometimes. Maybe a problem with API calls? Also the changing LAC problem could be result from wrong values. How can we get this fixed?

**Comment by E3V3A** [comment]
1. Hmm, yes, I've never seen "G" either, at least not on any Androids, that I can recall. 
2. I'm not sure about the missing digits, because going into *ServiceMode* menus, the current connection show the same LAC/CIDs. It could be a bug in the baseband FW though, that presents the wrong info. (It's very buggy!) The SM was behaving very strange when in 2G-only, where many other things didn't work, that works fine in 3G. Now this was tested on the GT-I9195 and I wanna test this also on the other phones...

**Comment by SecUpwN** [comment]
@CellularPrivacy/celltower and @larsgrefer, is this Issue still valid after improvements of LAC/CID/PSC? I am asking you since the person who opened it back then has left our project for good.

**Comment by jamieknotts0618** [comment]
I have a CELL SPY CATCHER APP!My phone started going crazy with all these number and some of it same and some different and the cid,nid,sid would have numbers and some dont that has the same MCC,MNC.
Can you tell me if that means somebody hacking into my phone or what's going on!
